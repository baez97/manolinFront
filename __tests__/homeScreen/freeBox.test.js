import React from 'react';
import renderer from 'react-test-renderer';
import FreeBox from '../../components/homeComponents/changes/freeBox';

jest.unmock('react-native')
jest.unmock('react')
jest.unmock('react-test-renderer')
jest.unmock('expo');

describe("ChangeBox", () => {
    describe("Methods", () => {
        it("HandleLongPress works properly", () => {
            const mockedProps = {
                freeOnPress : jest.fn(),
            }

            const fB = new FreeBox(mockedProps);
            fB.handleLongPress();
            expect( mockedProps.freeOnPress ).toBeCalled();
        });
    });

    describe("View", () => {
        it("Matches the Snapshot", () => {
            const mockedChange = {
                owner: "Ana",
                purposals: [],
                turn: "M",
                day: 2,
                month: 2,
                year: 2019,
                weekday: 2,
                type: "free",
                accepted: false
            }

            const view = <FreeBox change={mockedChange} />;
            const rendered = renderer.create(view);
            const renderedJson = rendered.toJSON();

            expect( renderedJson ).toMatchSnapshot();
        })
    })
});