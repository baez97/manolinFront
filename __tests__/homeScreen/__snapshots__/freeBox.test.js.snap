// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeBox View Matches the Snapshot 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "elevation": 15,
      "opacity": 1,
    }
  }
>
  <ExponentLinearGradient
    colors={
      Array [
        4287123141,
        4281607599,
      ]
    }
    endPoint={
      Array [
        0.8,
        1,
      ]
    }
    startPoint={
      Array [
        0.7,
        0,
      ]
    }
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 20,
        "elevation": 10,
        "flexDirection": "row",
        "height": 177,
        "justifyContent": "space-between",
        "marginBottom": 0,
        "marginLeft": 17.5,
        "marginRight": 17.5,
        "marginTop": 17.5,
        "paddingBottom": 35,
        "paddingLeft": 52.5,
        "paddingRight": 70,
        "paddingTop": 35,
        "width": 712.5,
      }
    }
  >
    <View>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "white",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 38.95833442608516,
            "textAlign": "left",
          }
        }
      >
        Martes 2 de Febrero
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": 336.5,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Object {
              "color": "#cfbcf9",
              "fontFamily": "montserrat-extra-bold",
              "fontSize": 31.87500089407465,
              "marginTop": 8,
              "textAlign": "left",
            }
          }
        >
          Ana
        </Text>
        <Text
          allowFontScaling={false}
          style={
            Object {
              "color": "#cfbcf9",
              "fontFamily": "montserrat-extra-bold",
              "fontSize": 31.87500089407465,
              "marginTop": 8,
              "textAlign": "left",
            }
          }
        >
          Librar
        </Text>
      </View>
    </View>
    <Text
      allowFontScaling={false}
      style={
        Object {
          "color": "white",
          "fontFamily": "montserrat-extra-bold",
          "fontSize": 106.25000298018863,
          "marginTop": -5,
          "textAlign": "center",
        }
      }
    >
      M
    </Text>
  </ExponentLinearGradient>
</View>
`;
